### PR TITLE
[8.0][product_custom_info] New module to have custom properties in products

### PR DESCRIPTION
**Pull request by yajo**
Continuing from #120, this is only the product part, now that the base part is in https://github.com/OCA/server-tools/pull/313.

It allows to add custom attributes to products.

Obviously, this will fail until https://github.com/OCA/server-tools/pull/313 is merged.

@rafaelbn @pedrobaeza @carlos-incaser.

### CODE REVIEW

File: oca_dependencies.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+server-tools
```
Comment (pedrobaeza) [inline oca_dependencies.txt]: Put here
```
server-tools https://github.com/Yajo/server-tools custom-info
```
and you'll be able to test this PR while the other PR is without being merged.

File: product_custom_info/__init__.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+# © 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# © 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from . import models
```
Comment (pedrobaeza) [inline product_custom_info/__init__.py]: Line at the end

File: product_custom_info/README.rst [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+===================
+Product Custom Info
+===================
+
+This module extends products to allow having custom properties on them.
+
+Usage
```
Comment (pedrobaeza) [inline product_custom_info/README.rst:23]: Add installation section indicating the dependency on OCA/server-tools

File: product_custom_info/models/product_custom_info.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+# © 2015 Antiun Ingeniería S.L. - Sergio Teruel
+# © 2015 Antiun Ingeniería S.L. - Carlos Dauden
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models
+
+
+class ProductTemplate(models.Model):
+    _name = "product.template"
+    _inherit = ["product.template", "custom.info"]
```
Comment (pedrobaeza) [inline product_custom_info/models/product_custom_info.py]: You should also inherit properties creating a new table, or this won't work when having multiple models that uses these tables.

Comment (Yajo) [inline product_custom_info/models/product_custom_info.py]: Not. See https://github.com/OCA/server-tools/pull/313#discussion_r48032780.

**Comment by Yajo** [comment]
Well, that line I added in https://github.com/Yajo/product-attribute/commit/fac5eb89e85d6c721187f5ce9dee9548f2e59091 seems ignored in runbot, so not quite useful... Is that expected?

**Comment by yajo** [comment]
WIP no more, https://github.com/OCA/server-tools/pull/313 is merged.

I rebased and removed fac5eb89e85d6c721187f5ce9dee9548f2e59091. Let's see if it gets :green_heart:

File: product_custom_info/README.rst [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+===================
+
+This module extends products to allow having custom properties on them.
+
+Installation
+============
+
+* Install `OCA/server-tools <https://github.com/OCA/server-tools>`_.
+
+Usage
```
Comment (pedrobaeza) [inline product_custom_info/README.rst:23]: It lacks section Configuration to indicate where you create the Custom Info Templates.

File: product_custom_info/README.rst [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+===================
+Product Custom Info
+===================
+
+This module extends products to allow having custom properties on them.
+
+Installation
+============
+
+* Install `OCA/server-tools <https://github.com/OCA/server-tools>`_.
```
Comment (pedrobaeza) [inline product_custom_info/README.rst]: Indicate the exact module it depends.

**Comment by yajo** [comment]
This one should be ready.

**Comment by pedrobaeza** [comment]
I see that I cannot create by hand templates to test this, because model fields are readonly. The problem should be in base_custom_info.

**Comment by yajo** [comment]
Well, it's the expected behavior. It's readonly to avoid users changing model of template and breaking things. You are expected to manage templates from the product form. Please follow usage instructions and you'll see it works fine.

**Comment by yajo** [comment]
... and in any case, if you consider that a bug, it's a discussion for `base_custom_info` as you pointed.

**Pull request closed without merging**